flang/lib/Lower/ConvertVariable.cpp

```diff
@@ -1650,14 +1650,9 @@ void Fortran::lower::mapSymbolAttributes(
 
 void Fortran::lower::defineModuleVariable(
     AbstractConverter &converter, const Fortran::lower::pft::Variable &var) {
-  // linkOnce linkage allows the definition to be kept by LLVM
-  // even if it is unused and there is not init (other than undef).
-  // Emitting llvm :'@glob = global type undef' would also work, but mlir
-  // always insert "external" and removes the undef init when lowering a
-  // global without explicit linkage. This ends-up in llvm removing the
-  // symbol if unsued potentially creating linking issues. Hence the use
-  // of linkOnce.
-  auto linkOnce = converter.getFirOpBuilder().createLinkOnceLinkage();
+  // Use empty linkage for module variables, which makes them available
+  // for use in another unit.
+  auto linkage = converter.getFirOpBuilder().getStringAttr("");
```

Comment:
Can we call this `externalLinkage` or something?

Also, I think you can abbreviate this line and make it bit more clear as well.
```c++
mlir::StringAttr externalLinkage;
```

Reply:
Done! Thanks for pointing this out!